drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -499,6 +500,11 @@ static int nrf5_tx(const struct device *dev,
 	nrf5_radio->tx_psdu[0] = payload_len + NRF5_FCS_LENGTH;
 	memcpy(nrf5_radio->tx_psdu + 1, payload, payload_len);
 
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
```

Comment:
We should keep this file independent of OpenThread specific settings. What about adding some `CONFIG_IEEE802154_TIME_SYNC` option which is enabled somewhere else when `CONFIG_OPENTHREAD_TIME_SYNC` is enabled?